File: 백준/Silver/1337. 올바른 배열/올바른 배열.py
import sys
input = sys.stdin.readline
def check(arr):
    arr.sort()
    n = len(arr)
    start = 0
    max_cnt = 0

    for end in range(n):
        while arr[end]-arr[start]>=5:
            start +=1
        current_window = end-start+1
        max_cnt = max(max_cnt,current_window)
    return max(0,5-max_cnt)
n = int(input())
numbers = list(int(input())for _ in range(n))

print(check(numbers))

File: 백준/Silver/1337. 올바른 배열/올바른 배열.java
import java.io.*;
import java.util.*;

public class Main {
    public static int check(int[] arr){
        Arrays.sort(arr);
        int n = arr.length;
        int start = 0;
        int maxCnt = 0;
        for(int end =0;end<n;end++){
            while(arr[end]-arr[start]>=5){
                start++;
            }
            int currentWindow = end-start+1;
            maxCnt = Math.max(maxCnt,currentWindow);
        }
        return Math.max(0,5-maxCnt);
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int [] numbers = new int[n];
        for(int i = 0;i<n;i++){
            numbers[i] = Integer.parseInt(br.readLine());
        }
        System.out.println(check(numbers));
    }
}
